<launch>
  	<arg name="camera_name" default="Outdoor_Blimp" />
  	<!-- video_stream_provider can be a number as a video device or a url of a video stream -->
  	<arg name="video_stream_provider" default="0" />
  	<!-- set camera fps to -->
  	<arg name="set_camera_fps" default="10" />
  	<!-- set buffer queue size of frame capturing to -->
  	<arg name="buffer_queue_size" default="100" />
  	<!-- frames per second to query the camera for -->
  	<arg name="fps" default="10" />
  	<!-- frame_id for the camera -->
  	<arg name="frame_id" default="raspicam" />
  	<!-- By default, calibrations are stored to file://${ROS_HOME}/camera_info/${NAME}.yaml
  	To use your own fill this arg with the corresponding url, e.g.:
  	"file:///$(find your_camera_package)/config/your_camera.yaml" -->
    <arg name="camera_info_url" default="" />
  	<!-- flip the image horizontally (mirror it) -->
  	<arg name="flip_horizontal" default="false" />
  	<!-- flip the image vertically -->
  	<arg name="flip_vertical" default="false" />
    <!-- force width and height, 0 means no forcing -->
     <!--   <arg name="width" default="410"/> 
        <arg name="height" default="308"/> -->
         <arg name="width" default="640"/>
        <arg name="height" default="480"/>
  	<!-- if show a image_view window subscribed to the generated stream -->
	<arg name="visualize" default="false"/>

   
   	<!-- images will be published at /camera_name/image with the image transports plugins (e.g.: compressed) installed -->
   	<group ns="$(arg camera_name)">
	    <node pkg="video_stream_opencv" type="video_stream" name="$(arg camera_name)_stream" output="screen"> 
	    	<remap from="camera" to="temporary" />
	    	<param name="camera_name" type="string" value="$(arg camera_name)" />
	        <param name="video_stream_provider" type="string" value="$(arg video_stream_provider)" />
	        <param name="set_camera_fps" type="double" value="$(arg set_camera_fps)" />
	        <param name="buffer_queue_size" type="int" value="$(arg buffer_queue_size)" />
	        <param name="fps" type="double" value="$(arg fps)" />
	        <param name="frame_id" type="string" value="raspicam" />
	        <param name="camera_info_url" type="string" value="$(arg camera_info_url)" />
	        <param name="flip_horizontal" type="bool" value="$(arg flip_horizontal)" />
	        <param name="flip_vertical" type="bool" value="$(arg flip_vertical)" />
	        <param name="width" type="int" value="$(arg width)" />
	        <param name="height" type="int" value="$(arg height)" />
	    </node>

	    <node if="$(arg visualize)" name="$(arg camera_name)_image_view" pkg="image_view" type="image_view">
	    	<remap from="image" to="temporary" />
	    </node>
	</group>
    <node name="Outdoor_Blimp_Publisher" pkg="outdoor_blimp" type="publisher.py" output="screen">
    </node>
    <node name="Outdoor_Blimp_Controller" pkg="outdoor_blimp" type="controller.py" output="screen">
	<rosparam param = "yaw_kp">2</rosparam>
    	<rosparam param = "yaw_ki">0.05</rosparam>
    	<rosparam param = "yaw_kd">0.3</rosparam>
    	<rosparam param = "trans_x_kp">0</rosparam>
    	<rosparam param = "trans_x_ki">0</rosparam>
    	<rosparam param = "trans_x_kd">0</rosparam>
	<rosparam param = "trans_z_kp">0</rosparam>
    	<rosparam param = "trans_z_ki">0</rosparam>
    	<rosparam param = "trans_z_kd">0</rosparam>
    </node>

    <node name="OB_Cam_IMU" pkg="ob_cam_imu" type="ob_cam_imu_node.py" output="screen">
        <rosparam param = "frequency">200</rosparam>
    </node>
    <node name="world_to_imu_bosch_link" pkg="tf" type="static_transform_publisher" args="0.0 0.0 0.0 0.0 0.0 0.0 world imu_bosch_link 200" />
    <node name="imu_bosch_link_to_raspicam"
			pkg="tf" type="static_transform_publisher"  
                        args="0.0 0.0 0.0 0.0 0.0 0.0 imu_bosch_link raspicam 200" />     

</launch>
